File: Software-Praktikum/src/server/bo/ChatAccessBO.py
from .Businessobject import Businessobject

# Stellt da, welches Profil auf welchen Chatraum zugreifen kann
# Bei den Chaträumen wird dabei unter Gruppenchaträume und Zweierchaträume aufgeteilt
# E: Zweierchat
# G: Gruppenchat
# Gleichzeitig wird Chataccess dafür genutzt, die Mitglieder einer Gruppe zu finden

# @author [Ha Mi Duong](https://github.com/HamiDuong)

class ChatAccessBO (Businessobject):
    # Instanziierung des Objekts der Klasse ChatAccess
    def __init__(self):
        super().__init__()
        self.profile_id = None
        self.room = None
        self.chattype = None

    def set_profile_id(self, value):
        self.profile_id = value

    def get_profile_id(self):
        return self.profile_id

    def set_room(self, room):
        self.room = room

    def get_room(self):
        return self.room

    def set_chattype(self, chattype):
        self.chattype = chattype

    def get_chattype(self):
        return self.chattype

    def __str__(self):
        return "Id: {}, Profil {} kann in Raum {}, Type: {}".format(self.get_id(), self.get_profile_id(),
                                                                    self.get_room(), self.get_chattype())

    # Umwandlung eines Dictonary Eintrags in ein Message Objekt
    @staticmethod
    def from_dict(dictionary=dict()):
        obj = ChatAccessBO()
        obj.set_id(dictionary["id"])
        obj.set_profile_id(dictionary["profile_id"])
        obj.set_room(dictionary["room"])
        obj.set_chattype(dictionary["chattype"])
        return obj

File: Software-Praktikum/Frontend/lernapp/src/API/ChataccessBO.js
import BusinessObject from './BusinessObject';

/**
 * Stellt da, welches Profil auf welchen Chatraum zugreifen kann
 * Bei den Chaträumen wird dabei unter Gruppenchaträume und Zweierchaträume aufgeteilt
 * E: Zweierchat
 * G: Gruppenchat
 * Gleichzeitig wird Chataccess dafür genutzt, die Mitglieder einer Gruppe zu finden
 * 
 * @author [Ha Mi Duong](https://github.com/HamiDuong)
 */

export default class ChataccessBO extends BusinessObject {

    /**
     * 
     * @param {int} profile_id - Id einer ProfileBO
     * @param {int} room - Id einer ChatroomBO
     * @param {string} chattype - Art des Chatraums (Gruppen oder Zweier)
     */
    constructor(profile_id, room, chattype){
        super();
        this.profile_id = profile_id;
        this.room = room;
        this.chattype = chattype;
    }

    set_profile_id(id){
        this.profile_id = id;
    }

    get_profile_id(){
        return this.profile_id;
    }

    set_room(id){
        this.room = id;
    }

    get_room(){
        return this.room;
    }

    set_chattype(chattype){
        this.chattype = chattype;
    }

    get_chattype(){
        return this.chattype;
    }
 
    static fromJSON(chataccess){
        let result = [];

        if(Array.isArray(chataccess)){
            chataccess.forEach((c) => {
                Object.setPrototypeOf(c, ChataccessBO.prototype);
                result.push(c);
            })
        }else{
            let c = chataccess;
            Object.setPrototypeOf(c, ChataccessBO.prototype);
            result.push(c);
        }
        return result;
    }
}